### ISSUE TITLE
Font diff tool from nototools

### ISSUE THREAD

**Issue by chrissimpkins** [sf-playground/hackover#3]
@davelab6 : You might like the discussion in googlei18n/nototools#158

**Comment by chrissimpkins** [comment]
@jublo ^^

@davelab6 : Oh the time that we would have saved if we had simply done a bit more exploration of your repositories :) 

We want a visual font diff tool to collaboratively develop open source fonts.  The conversion to UFO allows for text diffs, but our experience was that the text diffs were too onerous to wade through in order to try to determine what changes were actually made.  A visual diff tool for the glyphs + an OpenType table diff tool are what you really need.

**Comment by davelab6** [comment]
Right. I also want such a tool for the same purpose :) The upcoming fonttools feaLib export methods are going to be great for the second part of this. While I would love to see more web developers using the UFO infrastructure that @graphicore has done with ufoJS, and also the community around github.com/nodebox/opentype.js is doing, the momentum in the type community is still strong with Python and a diff extension for TruFont (the UFO native libre font editor, and I think despite its young age already the best libre editor available :) seems more likely to be sustained

**Comment by chrissimpkins** [comment]
A separate issue and an aside--

@davelab6 : Do you have any experience with pyftmerge in the fontTools library or scripted merges via fontforge?  I am trying to come up with an approach that allows users to perform 1:1 glyph replacements in the Hack release sets using an alternate set of glyphs.  We would provide these alternate glyphs to begin, but my goal is to allow users to contribute their own variants that can be patched into the fonts and maintain these in a repository so that anyone may access them.  This will be a workaround for the lack of alternate glyph support in most source code text editors at the moment.

**Comment by chrissimpkins** [comment]
> TruFont

Have not come across this.  Searching for it now

**Comment by chrissimpkins** [comment]
https://github.com/trufont/trufont

How stable is the TruFont project at this stage?

**Comment by davelab6** [comment]
TruFont is not at 1.0.0, so it can not be considered stable. However, it
has all the big names in libre fonts contributing to it :)

pyftmerge is much superior to FontForge; I can no longer recommend the
latter at all. pyftmerge should be reliable, and if you find bugs, please
report them on the fonttools repo and I expect them to be fixed pretty fast
:)

**Comment by chrissimpkins** [comment]
@davelab6 thank you very much for all of the information!
